File: sla_customization/sla_customization/report/incident_ticketing_report/incident_ticketing_report.js
// Copyright (c) 2026, Jay Anjarlekar and contributors
// For license information, please see license.txt

frappe.query_reports["Incident Ticketing Report"] = {
        "filters": [
        {
            "fieldname": "territories",
            "label": __("Territories"),
            "fieldtype": "MultiSelect",
            "options" : []
        },
        {
            "fieldname": "customer_group",
            "label": __("Customer Group"),
            "fieldtype": "MultiSelect",
            "options": ["Individual", "Commercial", "Non Profit"] // Static example
        }
    ],
    "onload": function(report) {
        // Fetch the data manually
        console.log("in onload")
        frappe.db.get_list('Email Account', { pluck: 'name' }).then(data => {
            // Get the filter object
            let f = report.get_filter('territories');
            console.log(data,f)
            // Set the options and refresh the control
            f.set_data(data); 
            f.refresh();
        });
    }
};

File: sla_customization/sla_customization/report/incident_ticketing_report/incident_ticketing_report.py
# Copyright (c) 2026, Jay Anjarlekar and contributors
# For license information, please see license.txt

import frappe
from frappe import _


def execute(filters: dict | None = None):
	"""Return columns and data for the report.

	This is the main entry point for the report. It accepts the filters as a
	dictionary and should return columns and data. It is called by the framework
	every time the report is refreshed or a filter is updated.
	"""
	columns = get_columns()
	data = get_data(filters)

	return columns, data


def get_columns() -> list[dict]:
	"""Return columns for the report.

	One field definition per column, just like a DocType field definition.
	"""
	return [
		{
			"label": _("From Date"),
			"fieldname": "from_date",
			"fieldtype": "Date",
		},
		{
			"label": _("To Date"),
			"fieldname": "to_date",
			"fieldtype": "Date",
		},

	]


def get_data(filters) -> list[list]:
	if filter:
		frappe.log_error(filters)
	return [
		["Row 1", 1],
		["Row 2", 2],
	]
